### PR TITLE
[10.0][ADD] Module Purchase Minimum Amount

### PR DESCRIPTION
**Pull request by rgarnau**
This module allows you to block the validation of a Purchase Order providing the block reason 'Minimum Purchase Order Amount per Vendor'. It depends on #443. 

@jbeficent @darshan-serpent

This module is still 'Work in progress'.

### CODE REVIEW

**Review by JordiBForgeFlow — APPROVED** [review]
Technical and functional review :+1:

**Comment by JordiBForgeFlow** [comment]
@mreficent can you review?

File: oca_dependencies.txt [hunk 1]
```diff
@@ -4,3 +4,4 @@ account-invoicing
 server-tools
 stock-logistics-workflow
 sale-workflow
+purchase-workflow https://github.com/Eficent/purchase-workflow 10.0-purchase_order_block
```
Comment (LoisRForgeFlow) [inline oca_dependencies.txt]: Do not forget to remove this, I would put it into a different commit to be always aware.

File: purchase_minimum_amount/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+{
+    "name": "Purchase Minimum Amount",
+    "author": "Eficent, Odoo Community Association (OCA)",
+    "version": "10.0.1.0.0",
+    "website": "https://github.com/OCA/purchase-workflow",
+    "depends": [
+        'purchase_order_approval_block',
```
Comment (rousseldenis) [inline purchase_minimum_amount/__manifest__.py]: I think this is more correct to have explicit import from purchase

File: purchase_minimum_amount/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        string="Purchase Minimum Amount",
+        readonly=True,
+    )
+
+    @api.model
+    def _check_minimum_amount(self):
+        po_amt_block = self.env.ref('purchase_minimum_amount.'
+                                    'minimum_amount_block_reason',
+                                    raise_if_not_found=False)
+        under_min = self.amount_untaxed < self.minimum_po_amount
```
Comment (rousseldenis) [inline purchase_minimum_amount/models/purchase_order.py]: There is an inconsistency between decorator (model) and self usage.

Either you decorate with api.multi and iterate or ensure_one(), or you don't decorate (even if you come from the create() which is model decorated, 'po' is a recordset)

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by rgarnau** [comment]
@rousseldenis @lreficent changes applied

**Review by rousseldenis — APPROVED** [review]
Code review. LGTM

**Comment by JordiBForgeFlow** [comment]
@rgarnau https://github.com/OCA/purchase-workflow/pull/443 has been merged. Can you rebase this branch?

**Comment by rgarnau** [comment]
@jbeficent done!
